app/javascript/dashboard/components/widgets/conversation/ConversationBulkActions.vue

```diff
@@ -0,0 +1,288 @@
+          />
+        </div>
+        <div class="container">
+          <ul v-if="!selectedAgent">
+            <li class="search-container">
+              <div class="agent-list-search flex-between">
+                <fluent-icon icon="search" class="search-icon" size="16" />
+                <input
+                  ref="search"
+                  v-model="query"
+                  type="search"
+                  placeholder="Search"
+                  class="agent--search_input"
+                />
+              </div>
+            </li>
+            <li v-for="agent in filteredAgents" :key="agent.id">
+              <div class="agent-list-item" @click="assignAgent(agent)">
+                <thumbnail
+                  src="agent.thumbnail"
+                  :username="agent.name"
+                  size="22px"
+                  class="margin-right-small"
+                />
+                <span class="reports-option__title">{{ agent.name }}</span>
+              </div>
+            </li>
+          </ul>
+          <div v-else class="agent-confirmation-container">
+            <p>
+              {{
+                $t('BULK_ACTION.ASSIGN_CONFIRMATION_LABEL', {
+                  conversationCount: conversations.length,
+                })
+              }}
+              <strong>
+                {{ selectedAgent.name }}
+              </strong>
+            </p>
+            <div class="agent-confirmation-actions">
+              <woot-button
+                color-scheme="primary"
+                variant="smooth"
+                @click="goBack"
+              >
+                {{ $t('BULK_ACTION.GO_BACK_LABEL') }}
+              </woot-button>
+              <woot-button
+                color-scheme="primary"
+                variant="flat"
+                :is-loading="uiFlags.isUpdating"
+                @click="submit"
+              >
+                {{ $t('BULK_ACTION.ASSIGN_LABEL') }}
+              </woot-button>
+            </div>
+          </div>
+        </div>
+      </div>
+    </transition>
+  </div>
+</template>
+
+<script>
+import Thumbnail from 'dashboard/components/widgets/Thumbnail.vue';
+import { mapGetters } from 'vuex';
+
+export default {
+  components: {
+    Thumbnail,
+  },
+  props: {
+    conversations: {
+      type: Array,
+      default: () => [],
+    },
+    allConversationsSelected: {
+      type: Boolean,
+      default: false,
+    },
+  },
+  data() {
+    return {
+      showAgentsList: false,
+      query: '',
+      selectedAgent: null,
+    };
+  },
+  computed: {
+    ...mapGetters({
+      uiFlags: 'bulkActions/getUIFlags',
+      inboxes: 'inboxes/getInboxes',
+    }),
+    filteredAgents() {
+      return this.assignableAgents.filter(agent =>
+        agent.name.toLowerCase().includes(this.query.toLowerCase())
+      );
+    },
+    assignableAgents() {
+      return this.$store.getters['inboxAssignableAgents/getAssignableAgents'](
+        this.inboxes[0].id
```

Comment:
Would this create a bug if there are no inboxes? Initially when there are no inboxes. Can you check?

Reply:
If there are no inboxes, there wouldn't be any conversations correct? @nithindavid

But, makes sense to handle this anyway.

Reply:
We will update this API, so not important.